ln1_f4.py

```diff
@@ -8,86 +8,188 @@
+    mem_free, mem_cached = int(mem_console_data[9]), int(mem_console_data[12])
+    return {'mem_total': mem_total, 'mem_used': mem_used, 'mem_free': mem_free, 'mem_cached': mem_cached}
 
-###############################################################################################################
 
-if os.path.isfile("/home/lax/PycharmProjects/learn1/database_cron") == False:  # Adding new record into database
-    with open("/home/lax/PycharmProjects/learn1/database_cron", "w") as database_file:
-        sysinfo_database = {"0": {
-            "email_send_hour": 24
-        }}
-        database_file.write(json.dumps(sysinfo_database))
-
-with open("/home/lax/PycharmProjects/learn1/database_cron", "r") as database_file:
-    data = database_file.read().strip()
-sysinfo_database = json.loads(data)
-sysdb_totalrecords = len(sysinfo_database)
-sysinfo_database[str(sysdb_totalrecords)] = {
-    "record_time": systime_customformat,
-    "cpu_total": cpu_total,
-    "cpu_user": cpu_user,
-    "cpu_sys": cpu_sys,
-    "cpu_idle": cpu_idle,
-    "mem_total": mem_total,
-    "mem_used": mem_used,
-    "mem_free": mem_free,
-    "mem_cached": mem_cached,
-    "hdd_total": hdd_total,
-    "hdd_used": hdd_used,
-    "hdd_free": hdd_free
-}
-with open("/home/lax/PycharmProjects/learn1/database_cron", "w") as database_file:
-    database_file.write(json.dumps(sysinfo_database))
+def get_hdd_info():
+    hdd_console_data = get_sysresinfo('df -m --total')
+    hdd_total, hdd_used = int(hdd_console_data[50]), int(hdd_console_data[51])
+    hdd_free = int(hdd_console_data[52])
+    return {'hdd_total': hdd_total, 'hdd_used': hdd_used, 'hdd_free': hdd_free}
 
 ###############################################################################################################
-dynamic_html_table = """
+
+
+class SysinfoDatabase(object):
+    """ Current database structure
+    {'2016,05,19,01,05,05': {
+                            'cpu_user': 45.33,
+                            'cpu_sys': 17.22,
+                            'cpu_total': 60.11,
+                            ....
+                            'hdd_free':5952
+                            }
+    } """
+    def __init__(self, db_file_name):
+        self.db_file_name = db_file_name
+        if not os.path.isfile(self.db_file_name):
+            self.sysinfo_database = {}
+            with open(self.db_file_name, 'w') as self.database_file:
+                self.database_file.write(json.dumps(self.sysinfo_database))
+            self.db_is_empty = True
+
+        else:
+            self.db_is_empty = False
+            with open(self.db_file_name, 'r') as self.database_file:
+                self.sysinfo_database = self.database_file.read().strip()
+                self.sysinfo_database = json.loads(self.sysinfo_database)
+
+        self.db_index_keywords = self.sysinfo_database.keys()
+        self.db_index_keywords.sort(reverse=True)
+
+        self.lastrh = None  # Variable definitions
```

Comment:
lines 84-92 - there are not instance variables, that variables just related to specific functions and not shared for others. So make them just local for specific functions.